.github/workflows/tests.yml

```diff
@@ -0,0 +1,448 @@
+              OPEN_RUNTIMES_BUILD_COMMAND: ""
+              OPEN_RUNTIMES_START_COMMAND: "java -jar src/function/java-runtime-1.0.0.jar"
+            # C++ 17
+            - runtimes: cpp-17
+              RUNTIME: cpp-17
+              TEST_CLASS: Base
+              ENTRYPOINT: tests.cc
+              IMAGE: openruntimes/cpp:v3-17
+              ARCH: linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64
+              TEST_SCRIPT: tests.sh
+              OPEN_RUNTIMES_BUILD_COMMAND: ""
+              OPEN_RUNTIMES_START_COMMAND: "src/function/cpp_runtime"
+            # C++ 20
+            - runtimes: cpp-20
+              RUNTIME: cpp-20
+              TEST_CLASS: Base
+              ENTRYPOINT: tests.cc
+              IMAGE: openruntimes/cpp:v3-20
+              ARCH: linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64
+              TEST_SCRIPT: tests.sh
+              OPEN_RUNTIMES_BUILD_COMMAND: ""
+              OPEN_RUNTIMES_START_COMMAND: "src/function/cpp_runtime"
+            # Dotnet 7.0
+            - runtimes: dotnet-7.0
+              RUNTIME: dotnet-7.0
+              TEST_CLASS: Base
+              ENTRYPOINT: Tests.cs
+              IMAGE: openruntimes/dotnet:v3-7.0
+              ARCH: linux/amd64,linux/arm64
+              TEST_SCRIPT: tests.sh
+              OPEN_RUNTIMES_BUILD_COMMAND: ""
+              OPEN_RUNTIMES_START_COMMAND: "dotnet src/function/DotNetRuntime.dll"
+            # Dotnet 6.0
+            - runtimes: dotnet-6.0
+              RUNTIME: dotnet-6.0
+              TEST_CLASS: Base
+              ENTRYPOINT: Tests.cs
+              IMAGE: openruntimes/dotnet:v3-6.0
+              ARCH: linux/amd64,linux/arm64
+              TEST_SCRIPT: tests.sh
+              OPEN_RUNTIMES_BUILD_COMMAND: ""
+              OPEN_RUNTIMES_START_COMMAND: "dotnet src/function/DotNetRuntime.dll"
+    runs-on: ubuntu-20.04
+    name: ${{ matrix.runtimes }}
+    steps: 
+        - name: Clone Repository
+          uses: actions/checkout@v3.5.3
+        - name: Setup Docker Buildx
+          uses: docker/setup-buildx-action@v2.9.1
+        - name: Check Docker Buildx
+          run: docker buildx version
+        - name: Install PHP
+          uses: shivammathur/setup-php@2.25.4
+          with:
+            php-version: '8.0'
+        - name: Check PHP Version
+          run: php -v
+        - name: Install Dependencies
+          run: composer install
+        - name: Run Tests
+          run : |
+           RUNTIME="${{matrix.RUNTIME}}" ENTRYPOINT="${{ matrix.ENTRYPOINT }}" TEST_CLASS="${{ matrix.TEST_CLASS }}" OPEN_RUNTIMES_BUILD_COMMAND="${{ matrix.OPEN_RUNTIMES_BUILD_COMMAND }}" OPEN_RUNTIMES_START_COMMAND="${{ matrix.OPEN_RUNTIMES_START_COMMAND }}" sh ./tests.sh
+        - if: startsWith(github.ref, 'refs/tags/')
+          name: Login to DockerHub
+          uses: docker/login-action@v2.2.0
+          with:
+            username: ${{ secrets.DOCKER_USERNAME }}
+            password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
@christyjacob4 
Secrets to be set in github secrets are these. 
Should it be `DOCKER_USERNAME` and `DOCKER_PASSWORD` or something different.